components/autofill/src/sync/engine.rs

```diff
@@ -81,8 +81,10 @@ impl<T: SyncRecord + std::fmt::Debug> SyncEngine for ConfigSyncEngine<T> {
         let timestamp = inbound.timestamp;
         let num_incoming = inbound.changes.len() as u32;
         let tx = db.writer.unchecked_transaction()?;
-        let incoming_impl = self.storage_impl.get_incoming_impl();
-        let outgoing_impl = self.storage_impl.get_outgoing_impl();
+        // XXX - key will need to come in externally somehow.
+        let key = "to-be-passed-in-via-sync-manager?";
```

Comment:
I had this very question - what will the process of configuring sync-manager look like from the consumer's point-of-view, when they need to provide this key? And what does it mean for our efforts to avoid this component holding on to a key in memory for any significant length of time?

Reply:
> And what does it mean for our efforts to avoid this component holding on to a key in memory for any significant length of time?

That's a great question, and the only vague thoughts I have on this are:

* We will expect consumers to pass it in on every sync call - ie, there will not be a `set_key()` type API, but instead be a param to the `sync_now()` api (and I imagine that new param might be, say, an optional, generic set of key/values per engine, and this specific engine will insist on a key and fail if not supplied)
* We will manage the short lifetime in our process by similarly not "storing" it anywhere that lives longer than a single sync - eg, the "engine" in this component will be constructed each sync and isn't a long-lived object.

I really haven't thought about what we can do to ensure the key isn't sitting around in the heap even after it's been "discarded" by us, and haven't really thought about how to insist future refactoring doesn't break these assumptions, but would love ideas.